### PR TITLE
fix(op_crates/fetch): correct regexp for fetch header

### PR DESCRIPTION
**Pull request by Trott**
Fix bug in regular expression and make the regular expression more
strict.

In a string passed to new RegExp(), '[\t\s]' is identical to '[ts]' and
not `/[\t\s]/`. For that, the backslash needs to be escaped in the
string. Futhermore, `\t` is the tab character and is included in the
special regexp value `\s` so is unnecessary.

That would reduce the RegExp to new RegExp(`^${value}\\s*;?`) but
there's no point in matching 0 or more space characters followed by 0 or
one semi-colons as that will match no matter what follows `value`.

To make it more strict, require one of space, semicolon, or
end-of-string after value.

### CODE REVIEW

**Comment by Trott** [comment]
In order to write a test for this that fails with the current regexp but passes with the one here, I think we'd need a fetch with a form that returns a `Content-type` header like `multipart/form-datassssssstttt`. Perhaps that's an edge case that can be omitted from testing but if not, I'm happy to try to write the test, but I may need a gentle shove in the right direction.

**Comment by Trott** [comment]
For just general confirmation that this change is correct:

```javascript
// Original buggy function
function hasHeaderValueOf(s, value) {
  return new RegExp(`^${value}[\t\s]*;?`).test(s);
}

// Fixed function with this PR's change
function hasHeaderValueOfNew(s, value) {
  return new RegExp(`^${value}(?:[\\s;]|$)`).test(s);
}

console.log(
  hasHeaderValueOf('multipart/form-datatstststs', 'multipart/form-data')
); // Logs true. Oh no!

console.log(
  hasHeaderValueOfNew('multipart/form-datatstststs', 'multipart/form-data')
); // Logs false. Yay!
```

**Comment by bartlomieju** [comment]
> In order to write a test for this that fails with the current regexp but passes with the one here, I think we'd need a fetch with a form that returns a `Content-type` header like `multipart/form-datassssssstttt`. Perhaps that's an edge case that can be omitted from testing but if not, I'm happy to try to write the test, but I may need a gentle shove in the right direction.

Sounds good, you can add a new endpoint for `test_server` (in `test_util/src/lib.rs`) which such content type.

**Review by crypt096 — APPROVED** [review]

**Comment by Trott** [comment]
> Sounds good, you can add a new endpoint for `test_server` (in `test_util/src/lib.rs`) which such content type.

OK, test added. Thanks!

**Comment by Trott** [comment]
Whoops, let's try again, but this time after checking in the changes made by the format script....

File: cli/tests/unit/fetch_test.ts [hunk 1]
```diff
@@ -249,6 +249,27 @@ unitTest(
...
+
+    try {
+      const formData = await response.formData();
+      fail(
+        "Response.formData() didn't throw on a incorrect multipart content type",
+      );
+    } catch (e) {
+      assertEquals(e.message, "Invalid form data");
+      assert(e instanceof TypeError);
+    }
```
Comment (bartlomieju) [inline cli/tests/unit/fetch_test.ts:268]: Nitpick: you can use `assertThrowsAsync` instead of manually failing the test

**Review by bartlomieju — APPROVED** [review]
@Trott thanks for the PR, looks great!

File: cli/tests/unit/fetch_test.ts (hunk 1 above)
Comment (Trott) [reply cli/tests/unit/fetch_test.ts:268]: Sure, done.
